packages/aws-cdk-lib/aws-events-targets/lib/ecs-task.ts

```diff
@@ -277,13 +277,20 @@ export class EcsTask implements events.IRuleTarget {
   }
 
   private createEventRolePolicyStatements(): iam.PolicyStatement[] {
-    const policyStatements = [new iam.PolicyStatement({
-      actions: ['ecs:RunTask'],
-      resources: [this.taskDefinition.taskDefinitionArn],
-      conditions: {
-        ArnEquals: { 'ecs:cluster': this.cluster.clusterArn },
-      },
-    })];
+    const policyStatements = [
+      new iam.PolicyStatement({
+        actions: ['ecs:RunTask'],
+        resources: [this.taskDefinition.taskDefinitionArn],
+        conditions: {
+          ArnEquals: { 'ecs:cluster': this.cluster.clusterArn },
+        },
+      }),
+      new iam.PolicyStatement({
+        actions: ['ecs:TagResource'],
+        // eslint-disable-next-line @aws-cdk/no-literal-partition
+        resources: [`arn:aws:ecs:${this.cluster.env.region}:*:task/${this.cluster.clusterName}/*`],
+      }),
```

Comment:
I'm wondering if this breaks compatibility? I saw in the issue that `March 29, 2024 - Tagging Authorization will be turned on for all AWS accounts.` so maybe after March 29th this should be standard but till then should there be condition here that checks if the tag property is set then adds the policy?

Reply:
Thank you! 😄 

This won't break compatibility, only adding a permission to a role.

Reply:
Talked it over with paul and we're ok with merging, thanks again for the change